src/test/java/teammates/test/cases/action/InstructorFeedbackSubmissionEditSaveActionTest.java

```diff
@@ -567,6 +569,86 @@ public void testExecuteAndPostProcess() {
         // No tests since contrib qn can only be answered by students to own team members including self.
     }
 
+    @Test
+    public void testExecuteAndPostProcess_responsesForDuplicateIdSubmitted_errorReturned() {
```

Comment:
Make sure the test bypass the system before the fix and the data integrity is broken.

Reply:
Checked. The test does not bypass the system before the fix. It gives an error `Trying to create a Feedback Response that exists: agR0ZXN0chYLEhBGZWVkYmFja1F1ZXN0aW9uGB8M/instructor1@course1.tmt:Team 1.1</td></div>'"`

This error occurs from the logic level rather than from Action level. But it only occurs for responses that already exist in the DB. For new responses the response for duplicate recipient bypasses the system before the fix. 

I am adding new tests for new responses as well in the tests to make sure all scenarios are covered.